src/main/java/com/eleks/academy/whoami/core/state/ProcessingQuestion.java

```diff
@@ -39,13 +43,39 @@ public void ask(String player, PlayerAction.Action question) {
 		throw new GameException("Not implemented");
 	}
 
-	;
+	public void answer(String player, String value) {
+		if (turn.getCurrentTurn().get(0).getValue() != null) {
+			if (isAnswerState()) {
+				turn.action(player, value);
+				answers.add(value);
+				if (answers.size() == players.size() - 1) {
+					turn.makeTurn(players.values().stream().toList(), collectAnswers());
+					answers.clear();
+					updatePlayersState(this.getCurrentTurn(), players);
+				}
+			} else {
+				throw new GameException("You can't answer");
+			}
+		} else {
+			throw new GameException("No question for answer");
+		}
+	}
 
-	public void answer(String player, PlayerAction answer) {
-		throw new GameException("Not implemented");
+	private boolean isAnswerState() {
+		return players.entrySet()
```

Comment:
we need to take from turn information about who has to answer and who has to ask